modelservice/src/test/java/com/expedia/adaptivealerting/modelservice/service/DetectorServiceImplTest.java

```diff
@@ -137,13 +158,58 @@ public void testDeleteDetector() {
         verify(repository, times(1)).deleteByUuid(someUuidStr);
     }
 
+    @Test
+    public void testGetDetectorsToBeTrained() {
+        serviceUnderTest.getDetectorsToBeTrained();
+        verify(repository, times(1)).findByDetectorConfig_TrainingMetaData_DateNextTrainingLessThan(anyString());
+    }
+
+    @Test
+    public void testUpdateDetectorTrainingTime() {
+        val uuid = this.someUuid.toString();
+        val timestamp = DateUtil.toUtcDate("2020-07-15 20:00:00").toInstant().toEpochMilli();
+        serviceUnderTest.updateDetectorTrainingTime(uuid, timestamp);
+        verify(repository, times(1)).findByUuid(someUuid.toString());
+        verify(repository, times(1)).save(legalParamsDetector);
+    }
+
+    @Test
+    public void testUpdateDetectorTrainingTime_withPreviousTrainingInfo() {
+        fillDetectorConfigValues(legalParamsDetector);
+        val uuid = this.someUuid.toString();
+        val timestamp = DateUtil.toUtcDate("2020-07-20 20:00:00").toInstant().toEpochMilli();
+        serviceUnderTest.updateDetectorTrainingTime(uuid, timestamp);
+        verify(repository, times(1)).findByUuid(someUuid.toString());
+        verify(repository, times(1)).save(legalParamsDetector);
+    }
+
     private void initTestObjects() {
         this.someUuid = UUID.randomUUID();
         val mom = ObjectMother.instance();
         this.legalParamsDetector = mom.buildDetector();
         legalParamsDetector.setUuid(someUuid);
     }
 
+    private Detector getDetectorToUpdate() {
+        val mom = ObjectMother.instance();
+        val detectorToUpdate = mom.buildDetector();
+        detectorToUpdate.setUuid(someUuid);
+
+        return detectorToUpdate;
+    }
+
+    private void fillDetectorConfigValues(Detector detector) {
+        val trainingMetaData = new Detector.TrainingMetaData();
+        trainingMetaData.setCronSchedule("42 1 * * 3");
+        trainingMetaData.setDateLastTrained(DateUtil.toUtcDate("2020-07-15 20:00:00"));
+        trainingMetaData.setDateNextTraining(DateUtil.toUtcDate("2020-07-22 20:00:00"));
+        detector.getDetectorConfig().setTrainingMetaData(trainingMetaData);
+
+        val hyperParams = new HashMap<String, Object>();
+        hyperParams.put("hampel_n_signma", 4);
```

Comment:
"hampel_n_sigma"?

Reply:
No, `hampel_n_signma` is correct. I took it from sample detector info in prod, [here](https://adaptive-alerting.prod.monitoring.expedia.com/modelservice/api/v3/detectors/findByUuid?uuid=cbf5d81e-27f0-4ef8-8915-5495c46e6f8f).